### PR TITLE
SAA-147: Reorganise activity pay tables.

### PR DESCRIPTION
**Pull request by timharrison-moj**
Please do not merge - there is a matching change coming for the front-end to cater for these changes.

Changes include:
* Removed activity_pay_band table
* Added iep_level and pay_band into activity_pay
* Made the relationship one-to-many, so Activity now contains a List<ActivityPay>
* Renamed IEP to incentive level. 
* Updated entities, models, tests, and data to match.

target-data-model

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/job/CreateActivitySessionsJob.kt [hunk 1]
```diff
@@ -43,5 +43,5 @@ class CreateActivitySessionsJob(
   }
 
   private fun List<ActivitySchedule>.filterActivitySchedulesWithNoPreExistingInstance(day: LocalDate) =
-    this.filter { it.instances.none { it.sessionDate == day } }
+    this.filter { it.instances.none { scheduledInstance -> scheduledInstance.sessionDate == day } }
 }
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/job/CreateActivitySessionsJob.kt:47]: An it ->  was shadowing another it ->  (warnings in Kotlin plugin)

File: src/main/resources/migration/common/V1__baseline.sql [hunk 2]
```diff
@@ -171,7 +173,7 @@ CREATE TABLE attendance (
   posted                boolean,
   recorded_time         timestamp,
   recorded_by           varchar(100),
-  status                varchar(20), -- WAIT, CANC, COMP ?
+  status                varchar(20), -- SCH, CANC, COMP, PAID?
```
Comment (matt-crowson-moj) [inline src/main/resources/migration/common/V1__baseline.sql:176]: Perhaps update the docs on the Attendance model to reflect the possible values?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/model/ActivityPay.kt [hunk 3]
```diff
@@ -8,17 +8,14 @@ data class ActivityPay(
...
 
-  @Schema(description = "The incentive/earned privilege standard rate per session for this activity", example = "125")
-  val iepStandardRate: Int? = null,
-
-  @Schema(description = "The incentive/earned privilege enhanced rate per session for this activity", example = "150")
-  val iepEnhancedRate: Int? = null,
+  @Schema(description = "The earning rate for one half day session for someone of this incentive level and pay band (in pence)", example = "150")
+  val rate: Int? = null,
 
   @Schema(description = "Where payment is related to produced amounts of a product, this indicates the payment rate per pieceRateItems produced", example = "150")
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/model/ActivityPay.kt:20]: Add **(in pence)** to piece rate also?

**Review by matt-crowson-moj — DISMISSED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/model/ActivityPay.kt (hunk 3 above)
Comment (timharrison-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/model/ActivityPay.kt]: Done this now, Matt.

**Pull request merged**